DataFormats/CTPPSDigi/src/CTPPSPixelDigiCollection.cc

```diff
@@ -0,0 +1,70 @@
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigiCollection.h"
+
+#include <algorithm>
+
+void CTPPSPixelDigiCollection::put(Range input, unsigned int detID) {
+/// put in Digis of detID
+
+/// store size of vector before put
+  IndexRange inputRange;
+
+/// put in CTPPSPixelDigis from input
+  bool first = true;
+
+/// fill input in temporary vector for sorting
+  std::vector<CTPPSPixelDigi> temporary;
+  auto sort_begin = input.first;
+  auto sort_end = input.second;
+  for ( ;sort_begin != sort_end; ++sort_begin ) {
+    temporary.push_back(*sort_begin);
+  }
+  std::sort(temporary.begin(),temporary.end());
+
+/// iterators over input
+  auto begin = temporary.begin();
+  auto end = temporary.end();
+  for ( ;begin != end; ++begin ) {
```

Comment:
this looks like a complicated way to write

inputRange.first=container_.size()
container_.insert(std::end(container), std::begin(temporary), std::end(temporary));
inputRange.second=container_.size()-1